warewulf/warewulf3
# Merge development (3.8) into master

So git doesn't have a `git merge -s theirs` anymore, so the below was followed: 

- Ensure latest updates
```
git fetch
```
- Ensure our clone matches upstream
```
git reset --hard origin
```
- Create a merge point between master and development ignoring all conflicts by using `ours`.
```
git merge -s ours origin/development
```
- Create a temp branch
```
git branch branchTEMP
```
- Override contents of branchTEMP (master + single merge commit) with development branch
```
git reset --hard origin/development
```
- Soft reset (i.e. keep contents of working tree and index)
```
git reset --soft branchTEMP
```
- Update the merge commit from above with our current contents and index, effectively reversing the `ours` merge into a `theirs` commit.
```
git commit --amend
```
- Remove the temp branch
```
git branch -D branchTEMP
```
- See that HEAD (master) and development have no differences now
```
git diff HEAD origin/development
```
- Checkout the contents of HEAD to a new branch dev_master_merge, suitable for a PR
```
git checkout -b dev_master_merge
```
- Push the branch
```
git push origin dev_master_merge
```
